src/Dapr.Actors/Client/ActorProxyFactory.cs

```diff
@@ -32,7 +32,16 @@ public ActorProxyOptions DefaultOptions
         /// <summary>
         /// Initializes a new instance of the <see cref="ActorProxyFactory"/> class.
         /// </summary>
-        public ActorProxyFactory(ActorProxyOptions options = null, HttpClientHandler handler = null)
+        [Obsolete("Use the constructor that accepts HttpMessageHandler. This will be removed in the future.")]
+        public ActorProxyFactory(ActorProxyOptions options, HttpClientHandler handler)
+            : this(options, (HttpMessageHandler)handler)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ActorProxyFactory"/> class.
+        /// </summary>
+        public ActorProxyFactory(ActorProxyOptions options = null, HttpMessageHandler handler = null)
```

Comment:
The way that this API was design isn't very flexible/testable. `HttpClientHandler` isn't a more specific type than `HttpMessageHandler`. 

The API change in this PR is necessary to make this code testable, and is done in a way that's binary-compatible. It's safe to remove a default value for an optional parameter as long as you provide a *better* overload. Only users that were explicitly passing a `HttpClientHandler` will be impacted by this, and they could just choose to suppress the warning.